### ISSUE TITLE
Filter non-inference ops from profiler

### ISSUE THREAD

**Issue by Dhravidk** [TheJoshBrod/CGinS#12]
Profiler captures `torch.nn.functional.dropout` (and other RNG/training-only or metadata/plumbing ops), and CGinS then tries to generate LLM CUDA kernels for them even though they don’t help inference and can be nondeterministic/no-op under `model.eval()`. Add an op-classifier/allowlist in the profiling → per-op artifact stage to skip: dropout/RNG (`*dropout*`, `rand*`, `bernoulli`, `multinomial`, etc.), view/meta (`view/reshape/permute/transpose/squeeze/unsqueeze/expand/as_strided`), shape queries (`size/stride/numel`), and copy/cast/alloc (`to/_to_copy/contiguous/clone/copy_/empty/zeros/ones/full/arange`). Acceptance: running the repro in `eval()+no_grad()` produces no artifacts/kernels for skipped ops and matches PyTorch outputs. Make sure changes are reflected in README.MD

```py
self.emb = nn.Embedding(vocab_size, embed_dim)
self.fc1 = nn.Linear(embed_dim, hidden_dim)
self.ln = nn.LayerNorm(hidden_dim)
self.dropout = nn.Dropout(dropout)
self.fc2 = nn.Linear(hidden_dim, num_classes)

x = self.emb(input_ids).mean(dim=1)
x = self.fc1(x)
x = F.gelu(x)
x = self.ln(x)
x = self.dropout(x)
return self.fc2(x)
```

```json
{
  "project": "test7",
  "device": "cuda",
  "op_counts": {
    "torch.nn.functional.embedding": 1,
    "torch.nn.functional.linear": 2,
    "torch.nn.functional.gelu": 1,
    "torch.nn.functional.layer_norm": 1,
    "torch.nn.functional.dropout": 1
  }
}
```
